include/mqtt/will.hpp

```diff
@@ -29,13 +29,14 @@ class will {
      * @param qos
      *        mqtt::qos
      */
-    will(std::string topic,
-         std::string message,
+    template <typename Topic, typename Message>
+    will(Topic&& topic,
```

Comment:
If the parameter is always going to be passed, unmodified, to the member variable, then instead of having the parameter be a template parameter with a forwarding/universal reference, just have the parameter be mqtt::buffer.

That way, if the data type can automatically be converted into an mqtt::buffer, it will, and if it can't, then the user will get a compiler error and know they have to fix it.

The addition of so many templates to functions is probably related to why the compilation time increased.

Reply:
The same reason as  https://github.com/redboltz/mqtt_cpp/pull/337#discussion_r313697374.

Maybe there is better way...

How about user defined literals ? https://wandbox.org/permlink/QSbOH0WV6OjocrbP

Reply:
Maybe https://github.com/redboltz/mqtt_cpp/pull/337#discussion_r313709762 is better solution.

Reply:
Oh, I did not see your comment here when I proposed using user defined literals.

Reply:
Fixed using user defined literals.